packages/flutter/lib/src/rendering/flex.dart

```diff
@@ -12,6 +12,84 @@ import 'layer.dart';
 import 'layout_helper.dart';
 import 'object.dart';
 
+// A type that describes a 2D size along the main axis and the cross axis of a
+// [RenderFlex].
+extension type const _AxisSize(Size _size) {
+  _AxisSize.fromAxisExtents({ required double mainAxisExtent, required double crossAxisExtent }) : this(Size(mainAxisExtent, crossAxisExtent));
+  _AxisSize.fromSize({ required Size size, required Axis direction }) : this(_convert(size, direction));
+
+  static Size _convert(Size size, Axis direction) {
+    return switch (direction) {
+      Axis.horizontal => size,
+      Axis.vertical => Size(size.height, size.width),
+    };
+  }
+  double get mainAxisExtent => _size.width;
+  double get crossAxisExtent => _size.height;
+
+  Size toSize(Axis direction) => _convert(_size, direction);
+}
+
+typedef _AscentDescent = ({double ascent, double descent});
+
+// The intermediate layout result when determining the overall size of the
+// RenderFlex. The ascent/descent values and the axis size are stored separately
+// because non-baseline aligned children are start-aligned.
+extension type const _LayoutAxisDimensions(({_AxisSize axisSize, _AscentDescent? ascentDescent}) axisDimensions) {
+  _AxisSize get axisSize => axisDimensions.axisSize;
+  _AscentDescent? get ascentDescent => axisDimensions.ascentDescent;
+
+  double get mainAxisExtent => axisSize.mainAxisExtent;
+  double get crossAxisExtent => axisSize.crossAxisExtent;
+
+  static const _LayoutAxisDimensions empty = _LayoutAxisDimensions((axisSize: _AxisSize(Size.zero), ascentDescent: null));
+
+  // Updates the `current` _LayoutAxisDimensions with the dimensions of the given
+  // child and returns the updated _LayoutAxisDimensions.
+  _LayoutAxisDimensions update(_AxisSize childSize, BaselineOffset baseline) {
```

Comment:
Every call site seems to require wrapping a double in BaselineOffset right before calling this method only for this method to unwrap it as the first step. Why not just pass the double in directly and avoid the mental indirection?